cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -175,8 +181,31 @@ def index_item(item, skip_index=False, groups_usage_info=None):
                 return
 
             item_content_groups = None
+
+            if item.category == "split_test":
+                user_partitions = item.user_partitions
+                split_partitions = []
+                if user_partitions:
```

Comment:
If the item is a split_test, then it has a ``user_partition_id`` which tells you which user partition it is associated with. I think you can just use ``item.description.get_selected_partition()`` to get the partition in question. You don't need to iterate over all the user partitions in the course like this.